web/core/components/modules/analytics-sidebar/root.tsx

```diff
@@ -287,16 +268,18 @@ export const ModuleAnalyticsSidebar: React.FC<Props> = observer((props) => {
 
   return (
     <div className="relative">
-      <LinkModal
+      <CreateUpdateModuleLinkModal
         isOpen={moduleLinkModal}
         handleClose={() => {
           setModuleLinkModal(false);
-          setSelectedLinkToUpdate(null);
+          const timeoutId = setTimeout(() => {
```

Comment:
@aaryan610 Not so sure about this code, can you please explain why is it being cleared after a timer and why we are clearing this timeout after its use?

Reply:
The input fields clear out even before the modal is closed, which gives a sense of flicker to the user. Hence, clearing them after the modal has closed.

Reply:
Oh ok, but why do you need to clear the timeout?